source/non_matching/quadrature_generator.cc

```diff
@@ -1489,6 +1558,38 @@ namespace NonMatching
 
 
 
+      template <int dim, typename VectorType>
+      void
+      RefSpaceFEFieldFunction<dim, VectorType>::set_subcell(
+        const std::vector<unsigned int> &mask,
+        const BoundingBox<dim> &         subcell_box)
+      {
+        for (unsigned int i = 0; i < local_dof_values_subcell.size(); ++i)
+          local_dof_values_subcell[i] = local_dof_values[renumber[mask[i]]];
```

Comment:
Please help me understand: The array `renumber` contains the inverse lexicographic numbering via line 1534 (`renumber = shape_info.lexicographic_numbering;`) from shape functions, so we may select an index in lexicographic order via `mask[i]` and then get it returned in the hierarchic order for `local_dof_values` that then maps back to the right subcell value. Did you verify this for `degree = 3`, just to be sure that all indices are translated correctly?

Reply:
Yes, I verified it for `degree = 3` and `dim = 2`. Do you want me to add `degree = 3` to the test?

Reply:
No, I don't think that is strictly necessary.